### PR TITLE
use _fit_linear_regression_xr in train_lt

### PR DESCRIPTION
**Pull request by mathause**
EDIT: This PR is now split in two - #142 will have to be merged first. I updated the description here.

---

This PR

- uses `_fit_linear_regression_xr` in `train_lt`, which (IMO) simplifies the function a bit

To do this I had to re-write (and currently duplicate) `train_l_prepare_X_y_wgteq` to return `xr.DataArray` objects. But I think this is only an intermittent step.

Unfortunately, `train_gt_ic_OLSVOLC` uses `fit_intercept=False` which `_fit_linear_regression_xr` cannot handle atm.

https://github.com/MESMER-group/mesmer/blob/f63fbeb0d9d1318a575e2aa7ad0febc4e675e98f/mesmer/calibrate_mesmer/train_gt.py#L264

@znicholls

### CODE REVIEW

File: mesmer/calibrate_mesmer/train_lt.py [hunk 1]
```diff
@@ -137,7 +136,8 @@ def train_lt(preds, targs, esm, cfg, save_params=True):
 
     # prepare predictors and targets such that they can be ingested into the training
     # function
-    X, y, wgt_scen_eq = train_l_prepare_X_y_wgteq(preds, targs)
+
```
Comment (mathause) [inline mesmer/calibrate_mesmer/train_lt.py:139]:
```suggestion
```

File: mesmer/core/linear_regression.py [hunk 2]
```diff
@@ -231,15 +231,19 @@ def linear_regression(
     # split `out` into individual DataArrays
     keys = ["intercept"] + list(predictors)
     dataarrays = {key: (target_dim, out[:, i]) for i, key in enumerate(keys)}
-    out = xr.Dataset(dataarrays, coords=target.coords).drop_vars(dim)
+
+    out = xr.Dataset(dataarrays, coords=target.coords)
+
```
Comment (mathause) [inline mesmer/core/linear_regression.py:236]:
```suggestion
```

**Review by znicholls — APPROVED** [review]
Very nice, cleans things up well

**Comment by znicholls** [comment]
Sorry should have done comment rather than approve given we're still at draft stage. 

> Unfortunately, train_gt_ic_OLSVOLC uses fit_intercept=False which _fit_linear_regression_xr cannot handle atm

Unfortunate but hopefully an easy fix ?

File: mesmer/calibrate_mesmer/train_lt.py [hunk 3]
```diff
@@ -10,10 +10,9 @@
 import os
 
 import joblib
-import numpy as np
-from sklearn.linear_model import LinearRegression
 
-from mesmer.calibrate_mesmer.train_utils import train_l_prepare_X_y_wgteq
+from mesmer.calibrate_mesmer.train_utils import _train_l_prepare_X_y_wgteq_xr
```
[the remark this answers is not in the document]
Comment (mathause) [reply mesmer/calibrate_mesmer/train_lt.py:14]: :laughing: yes good point - maybe something like `_stack_scen_and_ens_xr`. I do hope to eventually replace the function but a good name is always :+1:

**Comment by mathause** [comment]
Thanks for the quick look! 

> > Unfortunately, train_gt_ic_OLSVOLC uses fit_intercept=False which _fit_linear_regression_xr cannot handle atm
> 
> Unfortunate but hopefully an easy fix ?

I hope so, but I plan to do it in another PR, as it will probably make the logic more complicated.

**Comment by mathause** [comment]
@leabeusch: no worries - I don't intend to change the science (without good reason) but I will need to update some code before I can update that part of your original code

File: mesmer/calibrate_mesmer/train_lt.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mathause) [reply mesmer/calibrate_mesmer/train_lt.py:14]: @leabeusch this is now being restructured in #143 - I hope the new names are as understandable as the old one for you :pray:

**Comment by mathause** [comment]
Alrighty, this is ready & after #142 and #143 it's actually reasonably small.

File: mesmer/calibrate_mesmer/train_lt.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (leabeusch) [reply mesmer/calibrate_mesmer/train_lt.py]: Congrats on generalizing the function names beyond them only being understandable by me, without loosing their understandability for me. :smile:

**Review by znicholls — APPROVED** [review]
Nice

**Comment by mathause** [comment]
Thanks for the review!

**Comment by znicholls** [comment]
Smart

**Pull request merged**